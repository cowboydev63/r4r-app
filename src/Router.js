import React from 'react';
import { ConnectedRouter as Router } from 'react-router-redux';
import {
    Switch,
    Route
} from 'react-router-dom';
import Home from './views/home';
import Results from './views/results';
import Resource from './views/resource';
import PageNotFound from './components/PageNotFound';
import ErrorBoundary from './ErrorBoundary';

const AppRouter = ({ history }) => (
    <Router history={ history }>
        <ErrorBoundary>
            <Switch>
                <Route exact path="/" component={ Home } />
                <Route path="/search" component={ Results } />
                <Route path="/resource/:id" component={ Resource } />
                <Route path="*" component={ PageNotFound } />
            </Switch>
        </ErrorBoundary>
    </Router>
)

export default AppRouter;