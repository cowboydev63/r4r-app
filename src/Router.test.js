import React from 'react';
import { shallow } from 'enzyme';
import { Route } from 'react-router-dom';
import Router from './Router';
import Home from './views/home';
import Results from './views/results';
import Resource from './views/resource';

it('renders without crashing', () => {
  shallow(<Router />);
});

// Until I figure out testing dynamic routing in a shallow render, this will demonstrate a 
// simpler test without a params id or asterix.
it('renders the correct static route', () => {
  const wrapper = shallow(<Router />);
  const pathMap = wrapper.find(Route).reduce((pathMap, route) => {
    const routeProps = route.props();
    pathMap[routeProps.path] = routeProps.component;
    return pathMap;
  }, {});

  expect(pathMap['/r4r/search']).toBe(Results);
});