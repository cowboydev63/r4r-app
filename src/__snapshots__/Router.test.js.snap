// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router Container renders correctly 1`] = `
<ConnectedRouter
  history={Object {}}
>
  <Connect(ErrorBoundary)>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/search"
      />
      <Route
        component={[Function]}
        path="/resource/:id"
      />
      <Route
        component={[Function]}
        path="*"
      />
    </Switch>
  </Connect(ErrorBoundary)>
</ConnectedRouter>
`;
